R/process.R

```diff
@@ -430,44 +434,79 @@ process_generate_id <- function(x, prefix, sort = FALSE) {
 #'
 #' @param my_list List of input information
 #' @param dataset_id Identifier for a particular study in the AusTraits database
+#' @param traits Table of trait data (for this function, just the data.csv file with custom_R_code applied)
 #' @return Tibble with context details if available
 #' @importFrom rlang .data
 #'
 #' @examples
 #' \dontrun{
-#' process_format_contexts(read_metadata("data/Apgaua_2017/metadata.yml")$context)
+#' process_format_contexts(read_metadata("data/Apgaua_2017/metadata.yml")$context, dataset_id, traits)
 #' }
-process_format_contexts <- function(my_list, dataset_id) {
+process_format_contexts <- function(my_list, dataset_id, traits) {
 
-  f <- function(x) {
+  f <- function(x, id) {
     tibble::tibble(
     context_property = x$context_property,
     category = x$category,
     var_in = x$var_in,
-    util_list_to_df2(x$values))
+    util_list_to_df2(x$values)
+    )  %>%
+    dplyr::mutate(dataset_id = dataset_id) %>%
+    dplyr::select(dplyr::any_of(
+        c("dataset_id", "context_property", "category", "var_in",
+          "find", "value", "description"))
+      )
   }
-
+  
   if (!is.na(my_list[1])) {
-    contexts <-
+    
+      contexts <-
       my_list %>%
-      purrr::map_df(f) %>%
-      dplyr::mutate(dataset_id = dataset_id) %>%
-      dplyr::select(dplyr::any_of(
-        c("dataset_id", "context_property", "category", "var_in",
-        "find", "value", "description"))
+      purrr::map(f, dataset_id)
+    
+    for (i in 1:length(contexts)) {
+      
+      ## if the field `description` is missing from metadata[["contexts"]] for the specific context property, create a column now 
+      if (!"description" %in% names(contexts[[i]])) {
+        contexts[[i]]$description <- NA_character_
+      }
+      
+      ## if the fields `find` and `value` are both missing from metadata[["contexts"]] for the specific context property create them
+      ## they are both the unique set of values in the column in the data.csv file.
+      if(!"find" %in% names(contexts[[i]]) & 
+         !"value" %in% names(contexts[[i]])) {
+        to_join <- unique(traits[[contexts[[i]]$var_in[1]]]) %>%
```

Comment:
The `[1]` in this line suggests it will only sue the first instance of `var_in`, is that intended?